src/hooks/useInitialStartup.js

```diff
@@ -114,36 +114,36 @@ export const useInitialStartup = () => {
 
   const getDiscoveryData = async () => {
     try {
-      getFollowing().then((response) => {
-        following.setFollowingUsers(response.data, followingDispatch);
-      });
-
-      getFollowedDomain().then((response) => {
-        following.setFollowingDomain(response.data.data, followingDispatch);
-      });
-
-      getFollowingTopic().then((response) => {
-        following.setFollowingTopics(response.data, followingDispatch);
-      });
+      const discoveryInitialTopicResponse = await DiscoveryRepo.fetchInitialDiscoveryTopics();
+      DiscoveryAction.setDiscoveryInitialTopics(
+        discoveryInitialTopicResponse.suggestedTopics,
+        discoveryDispatch
+      );
 
       const discoveryInitialUserResponse = await DiscoveryRepo.fetchInitialDiscoveryUsers();
       DiscoveryAction.setDiscoveryInitialUsers(
         discoveryInitialUserResponse.suggestedUsers,
         discoveryDispatch
       );
 
-      const discoveryInitialTopicResponse = await DiscoveryRepo.fetchInitialDiscoveryTopics();
-      DiscoveryAction.setDiscoveryInitialTopics(
-        discoveryInitialTopicResponse.suggestedTopics,
-        discoveryDispatch
-      );
-
       const discoveryInitialDomainResponse = await DiscoveryRepo.fetchInitialDiscoveryDomains();
       DiscoveryAction.setDiscoveryInitialDomains(
         discoveryInitialDomainResponse.suggestedDomains,
         discoveryDispatch
       );
 
+      getFollowingTopic().then((response) => {
+        following.setFollowingTopics(response.data, followingDispatch);
+      });
+
+      getFollowing().then((response) => {
+        following.setFollowingUsers(response.data, followingDispatch);
+      });
+
+      getFollowedDomain().then((response) => {
+        following.setFollowingDomain(response.data.data, followingDispatch);
+      });
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are primarily reordering the sequence of API calls and converting them to use `async/await` instead of promises. This is a good practice as it makes the code more readable and easier to understand. However, it's important to note that using `async/await` will cause the execution to wait for each request to complete before moving on to the next one. If these requests do not depend on each other, you could use `Promise.all()` to execute them concurrently, which could improve performance.

Here's an example of how you might implement this:

```diff
-  const discoveryInitialTopicResponse = await DiscoveryRepo.fetchInitialDiscoveryTopics();
-  DiscoveryAction.setDiscoveryInitialTopics(
-    discoveryInitialTopicResponse.suggestedTopics,
-    discoveryDispatch
-  );
-
-  const discoveryInitialUserResponse = await DiscoveryRepo.fetchInitialDiscoveryUsers();
-  DiscoveryAction.setDiscoveryInitialUsers(
-    discoveryInitialUserResponse.suggestedUsers,
-    discoveryDispatch
-  );
-
-  const discoveryInitialDomainResponse = await DiscoveryRepo.fetchInitialDiscoveryDomains();
-  DiscoveryAction.setDiscoveryInitialDomains(
-    discoveryInitialDomainResponse.suggestedDomains,
-    discoveryDispatch
-  );

+  const [discoveryInitialTopicResponse, discoveryInitialUserResponse, discoveryInitialDomainResponse] = await Promise.all([
+    DiscoveryRepo.fetchInitialDiscoveryTopics(),
+    DiscoveryRepo.fetchInitialDiscoveryUsers(),
+    DiscoveryRepo.fetchInitialDiscoveryDomains()
+  ]);
+
+  DiscoveryAction.setDiscoveryInitialTopics(
+    discoveryInitialTopicResponse.suggestedTopics,
+    discoveryDispatch
+  );
+
+  DiscoveryAction.setDiscoveryInitialUsers(
+    discoveryInitialUserResponse.suggestedUsers,
+    discoveryDispatch
+  );
+
+  DiscoveryAction.setDiscoveryInitialDomains(
+    discoveryInitialDomainResponse.suggestedDomains,
+    discoveryDispatch
+  );
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->